src/System.Memory/tests/Performance/Perf.Utf8Parser.cs

```diff
@@ -0,0 +1,232 @@
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData("42")] // standard parse
+        [InlineData("0")] // min value
+        [InlineData("255")] // max value
+        private static void ParserByte(string text)
+        {
+            byte[] utf8ByteArray = Encoding.UTF8.GetBytes(text);
+            ReadOnlySpan<byte> utf8ByteSpan = utf8ByteArray;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        Utf8Parser.TryParse(utf8ByteSpan, out byte value, out int bytesConsumed);
+                        TestHelpers.DoNotIgnore(value, bytesConsumed);
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData("4212")] // standard parse
+        [InlineData("-32768")] // min value
+        [InlineData("32767")] // max value
+        private static void ParserInt16(string text)
+        {
+            byte[] utf8ByteArray = Encoding.UTF8.GetBytes(text);
+            ReadOnlySpan<byte> utf8ByteSpan = utf8ByteArray;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        Utf8Parser.TryParse(utf8ByteSpan, out short value, out int bytesConsumed);
+                        TestHelpers.DoNotIgnore(value, bytesConsumed);
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData("4212")] // standard parse
+        [InlineData("0")] // min value
+        [InlineData("65535")] // max value
+        private static void ParserUInt16(string text)
+        {
+            byte[] utf8ByteArray = Encoding.UTF8.GetBytes(text);
+            ReadOnlySpan<byte> utf8ByteSpan = utf8ByteArray;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        Utf8Parser.TryParse(utf8ByteSpan, out ushort value, out int bytesConsumed);
+                        TestHelpers.DoNotIgnore(value, bytesConsumed);
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData("12837467")] // standard parse
+        [InlineData("-2147483648")] // min value
+        [InlineData("2147483647")] // max value
+        private static void ParserInt32(string text)
```

Comment:
What about adding baseline to compare performance against int.TryParse?

Reply:
I wasn't sure how much usage the perf tests in CoreFx got so I only added the most critical scenarios and mostly so we'd have a set place to add more. We can beef this up for sure - but it might be better to do it on an as-needed basis. There's also a lot of stuff in that test assembly.